modules/core/src/core/animation-loop.js

```diff
@@ -190,6 +197,13 @@ export default class AnimationLoop {
     return this;
   }
 
+  getCanvasDataURL() {
+    this.setNeedsRedraw('getCanvasDataUrl');
+    return new Promise(resolve => {
```

Comment:
Nit: Maybe overkill to check if called multiple times before resolved?

In that case, could just store the promise on the class, store the resolver, and return the promise if already created.

```
if (!this._screenCapturePromise) {
  this._screenCapturePromise = new Promise(resolve => {
    this._resolveScreenCapturePromise = resolve;
  });
}
return this._screenCapturePromise;
```

Reply:
Good call.